core/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -243,14 +248,35 @@ private void applyDeletedIndices(final ClusterChangedEvent event) {
                 indexSettings = idxService.getIndexSettings();
                 deleteIndex(index, "index no longer part of the metadata");
             } else {
-                final IndexMetaData metaData = previousState.metaData().getIndexSafe(index);
-                indexSettings = new IndexSettings(metaData, settings);
-                indicesService.deleteClosedIndex("closed index no longer part of the metadata", metaData, event.state());
+                IndexMetaData metaData = previousState.metaData().index(index);
+                if (metaData == null) {
+                    // The previous cluster state's metadata also does not contain the index,
+                    // which is what happens on node startup when an index was deleted while the
+                    // node was not part of the cluster.  In this case, try reading the index
+                    // metadata from disk.  If its not there, there is nothing to delete.
+                    try {
+                        metaData = metaStateService.loadIndexState(index);
+                    } catch (IOException e) {
+                        // any other error handling here?
+                        logger.warn("[{}] Deletion request, but unable to read index metadata from disk, " +
+                                    "likely scenario is an index was created and deleted while the node was offline.", e, index);
+                        metaData = null;
+                    }
+                }
+                // index metadata being null means there is nothing to delete
+                if (metaData != null) {
+                    indexSettings = new IndexSettings(metaData, settings);
+                    indicesService.deleteClosedIndex("closed index no longer part of the metadata", metaData, event.state());
+                } else {
+                    indexSettings = null;
+                }
```

Comment:
@bleskes I had to change the logic here, because in the case of a node restarting, its previous state will not contain the index metadata for an index that was deleted while it was offline, so if the index metadata for the deleted index (part of the tombstones in the cluster state) is not in the previous cluster state, I try to read it off disk.  I had to introduce the `MetaStateService` as a dependency in this class in order to do that.  If the index metadata could not also be read off of disk, that means the index was both created and deleted while the node was offline, so there is nothing to do.  

I am unsure if this is the best approach, so would appreciate your feedback.